opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -1103,10 +1103,7 @@ assignShutConnections(data::Wells& wsrpt,
 
     for (const auto& well : this->wells_ecl_) {
         auto& xwel = wsrpt[well.name()]; // data::Wells is a std::map<>
-
-        xwel.dynamicStatus = this->schedule()
-            .getWell(well.name(), reportStepIndex).getStatus();
-
+        xwel.dynamicStatus  = this->wellState().well(well.name()).status;
```

Comment:
This breaks the `CTFAC` summary vector, especially in cases using `WELPI`.  Please see PR #2866, commit f6130861d.

Reply:
> So the correct behavior is that CTFAC should be zero for dynamically shut wells.

Yes.  We altered this behaviour in b982ad0fd (PR #3093).

Reply:
I think the test failure for the THP case is an improvement as the well is only temporary opened for well testing but still it gives output.